### ISSUE TITLE
Arabic text: display of Ground Truth

### ISSUE THREAD

**Issue by andbue** [Calamari-OCR/calamari#239]
When training on Arabic data, building on the "arabic" model from calamari_models_experimental, in the console output 10 examples shown after each epoch, in the GT all characters are in the wrong order.

**Comment by andbue** [comment]
The problem is here: https://github.com/Calamari-OCR/calamari/blob/7dc84b498ab9e6b9279d4b597848c72bdc203322/calamari_ocr/ocr/model/model.py#L68-L78
sample.targets is in display order for Arabic text, i.e. the first char in the string is the one at the left. sample.outputs, however, has been through the post processing, i.e. bidi.get_display has turned all chars around. It seems to work just fine in the training process, but those outputs, especially the Levenshtein-results, are quite confusing for the user.

The best way to fix this would be either to apply the postprocessing to the targets before comparing and displaying them, maybe here:
https://github.com/Calamari-OCR/calamari/blob/7dc84b498ab9e6b9279d4b597848c72bdc203322/calamari_ocr/ocr/dataset/textprocessors/text_processor.py#L14-L29
The `elif` in L18 prevents the processing. Should L14 be something like `... and not "sentence" in targets`?

**Comment by ChWick** [comment]
maybe it should be a simple `if` in line 18 and line 21? I think case l24 should never occur, so:
```python
        if isinstance(outputs, Prediction):
            prediction: Prediction = outputs
            prediction.sentence = self._apply_single(prediction.sentence, meta)
        if isinstance(targets, dict) and "sentence" in targets:
            targets["sentence"] = self._apply_single(targets["sentence"], meta)
        if isinstance(outputs, dict) and "sentence" in outputs:
            outputs["sentence"] = self._apply_single(outputs["sentence"], meta)
        return sample
```

**Comment by andbue** [comment]
For some reason, your suggested fix totally breaks training of bidi models, it seems to turn around targets where they shouldn't be inverted.

What might be related: calamari-predict-and-eval does not work with Arabic text. Running calamari-predict and then calamari-eval works as intended, though.

**Comment by amitdo** [comment]
Does Calamari assumes the terminal does not support the bidi algorithm and thus always do rtl->ltr before displaying rtl (or mixed rtl/ltr) text? 

Gnome terminal supports BIDI since 3.33.3.

**Comment by andbue** [comment]
I don't think this is the problem here, since the display of the prediction seems to be correct. The problem is that postprocessing is not applied to the "targets", which means that the Ground Truth is sorted from left to right at this point. Running the postprocessing for targets in text_processor.py fixes the display issues but breaks training for RTL text.
